src/resty/ljsonschema/init.lua

```diff
@@ -23,6 +23,11 @@ local tonumber = tonumber
 local default_null = nil        -- default null token
 local default_array_mt = nil    -- default array_mt metatable
 local default_match_pattern     -- default reg-ex engine to use
+
+local function default_str_len(s)
+  return #s
+end
```

Comment:
for performance; can we NOT have a default, but instead if it is not given, then in the generated code inject `#whatever` ?